### PR TITLE
Fix item key lookup on updated items in scene

### PR DESCRIPTION
**Pull request by zzmp**
- Replace the Scene::_masterBucketMap with a single _masterNonspatialSet to complement the _masterSpatialTree
- Update shadow fetch/sort and clean old fetch/sort/cull code, including a new file, CullTask.h/cpp

### CODE REVIEW

**Comment by zzmp** [comment]
#### Intents

Overlays were broken in that they would not update their render pipeline when an edit caused their itemKey to change. This meant that if you changed an overlay to/from opaque or drawInFront, it would no longer render correctly. This PR is meant to fix that. It also cleaned up other rendering code, so there is a bit more to test.

https://app.asana.com/0/74051501495175/94607584879271

#### Testing

General look should be tested. Overlays with the mentioned properties toggled should be tested.

Shadows should be tested, with regard to what is actually casting a shadow in the scene (this shouldn't fail the QA, but is good to note); shadows now uses a different item fetching construct, so it will probably look slightly different.

I tested using the following script. Note that drawInFront overlay shaders are broken - they do not shade, and make the cubes in the testing script look red; this is documented in https://app.asana.com/0/74051501495175/95028771431057.

```
// To use, copy this into the console and enter it. Then sequentially run through the tests:
// run(0)
// run(1)
// ...
// run(12)

var overlay = 0;

commands = ["overlay = Overlays.addOverlay('cube', { position: { x: 8.1, y: 23.1, z: 17 }, solid: true });",
"Overlays.editOverlay(overlay, { alpha: 1 });",
"Overlays.editOverlay(overlay, { alpha: 0.6 });",
"Overlays.editOverlay(overlay, { drawInFront: true });",
"Overlays.editOverlay(overlay, { alpha: 1 });",
"Overlays.editOverlay(overlay, { alpha: 0.6 });",
"Overlays.editOverlay(overlay, { drawInFront: false });",
"Overlays.editOverlay(overlay, { alpha: 1 });",
"Overlays.editOverlay(overlay, { drawInFront: true });",
"Overlays.editOverlay(overlay, { drawInFront: false });",
"Overlays.deleteOverlay(overlay);",
"overlay = Overlays.addOverlay('cube', { position: { x: 8.1, y: 23.1, z: 17 }, solid: true, drawInFront: true });",
"Overlays.deleteOverlay(overlay);"]

function run(cmd) { eval(commands[cmd]); }
function test(num) { for (var i = 0; i < num; i++) run(i); }
```

**Comment by hifi-gustavo** [comment]
Build console output for commit: 32807012e8124ede2e94549cb6ec01933d8c3aff available at: https://gist.github.com/da9148287b54c124b516<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7218.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7218.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 7 - used demo code shown here.  Running through options showed what I think is correct behavior - didn't see anything break and everything else works proper.  QA Pass.

**Comment by zzmp** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: d1324b2e7a984b3a922a77384bd1ecf96318acc8 available at: https://gist.github.com/da9148287b54c124b516<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7218.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7218.dmg

File: libraries/render/src/render/SortTask.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
+//
+//  CullTask.cpp
```
Comment (samcake) [inline libraries/render/src/render/SortTask.cpp]: SO is that sortTask,cpp now ?

**Pull request merged**